### PR TITLE
Fix jdkSourceLevel

### LINKED ISSUE #27858: JspOption jdkSourceLevel Disabled Unintentionally
**Describe the bug**  
The jdkSourceLevel property was unintentionally disabled when the new javaSourceLevel option was introduced.  The option still exists via the metatype, but it's input is not honored.  It always uses Java 8 as the source. 

This only applies to JSPs that need their their code limited to Java 7 syntax or below.

**Steps to Reproduce**  
Set jdkSourceLevel to 17 (or lower) but the source option used during compilation is still Java 8. 

The JSPs will still compile, but will not fail when expected (i.e Lambdas in Java 7 source). Unlikely scenario, but the property is not doing what it should.

**Expected behavior**  
The JSP should compiled with the source level set according to value set through jdkSourceLevel.

**Diagnostic information:**  
 - OpenLiberty Version: 24.0.0.1, 24.0.0.2 and 24.0.0.3
 - Affected feature(s) jsp-2.2, jsp-2.3, pages-3.0, and pages-3.1
 - Java Version:  N/A
 - server.xml configuration:  `<jspEngine jdkSourceLevel="17"/>`  (jdkSourceLevel accepts values between 13 and 18) 

**Additional context**  
The jdkSourceLevel was marked internal so that it will not show up in the documentation, but it was not removed from the code. This was to prevent breaking changes, but this error unfortunately occurred. This documentation removed jdkSourceLevel so that new users will opt to use `javaSourceLevel`. 

jspEngine Doc:  https://openliberty.io/docs/latest/reference/config/jspEngine.html#

This only applies to the `source` option.  Per the javap -source description: `Provide source compatibility with the specified Java SE release`. 

This does not apply to `target` (not used in JSP).

### PR DESCRIPTION
**Pull request by volosied**
fixes #27858

### CODE REVIEW

File: dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -23,7 +23,7 @@ extendedDocumentRoot.name=Extended document root
 extendedDocumentRoot=Directory that the JSP engine will search for additional JSP files to serve.
 
 javaSourceLevel.name=Java source level
-javaSourceLevel=Specifies the default Java source level for JSPs that are compiled by the JSP engine. Accepts one of the following LTS releases as a value: 8, 11, 17, and 21.
+javaSourceLevel=Specifies the default Java source level for JSPs that are compiled by the JSP engine. Accepts one of the following LTS releases as a value: 8, 11, 17, and 21. Note: jdkSourceLevel is supported, but it's use is discouraged.
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties]: Not sure how to word this. Perhaps say "support still exists"?

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/bootstrap.properties [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-2.0/
+# 
+# SPDX-License-Identifier: EPL-2.0
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info:com.ibm.ws.jsp*=all
+
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/bootstrap.properties]: new line

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+        assertTrue("Compilation error message not found!", response.getText().contains("Lambda expressions are allowed only at source level 1.8 or above")); 
+    }
+
+    /*
+     * Ensure JdkSourceLevel works by testing option 17 (Java 7).
+     * Lambda were added in Java 8, so the JSP is expected to fail.
+     */
+    @Test
+    @AllowedFFDC("java.security.PrivilegedActionException") // Occurs in EE10 and lower
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:146]: What is the exception something in JSP we need to take a look at in the future or is this one expected?

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/server.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    <featureManager>
+        <feature>jsp-2.3</feature>
+        <feature>localConnector-1.0</feature>
+    </featureManager>
+
+    <!-- Note: jdkSourceLevel has been marked internal -->
+    <jspEngine jdkSourceLevel="17" prepareJSPs="0"/>
+
+    <logging traceSpecification="*=info=enabled:com.ibm.ws.jsp*=all" maxFileSize="32" maxFiles="24" traceFormat="BASIC"/>
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/server.xml]: We usually move the tracing to the bootstrap.properties.

Comment (pnicolucci) [reply dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/server.xml]: I see it is already there, but you'll want to update the other attributes you have defined here.

File: dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties]: I don't like referencing an `internal` metatype from a public one. We stopped documenting the `jdkSourceLevel` to promote the use of `javaSourceLevel`

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java (hunk 3 above)
Comment (volosied) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:146]:
```
Exception = java.security.PrivilegedActionException
Source = com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.findWrapper
probeid = 445
Stack Dump = java.security.PrivilegedActionException: com.ibm.ws.jsp.JspCoreException: JSPG0049E: /index.jsp failed to compile : 

JSPG0091E: An error occurred at line: 27 in the file: /index.jsp
JSPG0093E: Generated servlet error from file: /index.jsp 

/Users/volosied/libertyGit/open-liberty/dev/build.image/wlp/usr/servers/jspJava7Server/workarea/org.eclipse.osgi/76/data/temp/default_node/SMF_WebContainer/TestJSPWithJava8/TestJSPWithJava8/_index.java:139: error: lambda expressions are not supported in -source 7
          messages.stream().forEach(e -> write.print(e));
                                      ^
  (use -source 8 or higher to enable lambda expressions)

	at java.base/java.security.AccessController.doPrivileged(AccessController.java:573)
	at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.findWrapper(AbstractJSPExtensionProcessor.java:482)
	at com.ibm.ws.jsp.webcontainerext.ws.PrepareJspHelper$PrepareJspHelperThread.handleCompile(PrepareJspHelper.java:493)
	at com.ibm.ws.jsp.webcontainerext.ws.PrepareJspHelper$PrepareJspHelperThread.run(PrepareJspHelper.java:460)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.ibm.ws.jsp.JspCoreException: JSPG0049E: /index.jsp failed to compile : 
```

Not sure why it's a security exception, but the compilation is the cause.

File: dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/server.xml (hunk 4 above)
Comment (volosied) [reply dev/com.ibm.ws.jsp.2.3_fat/publish/servers/jspJava7Server/server.xml]: I removed the server.xml logging and will rely on the bootstrap.

**Comment by volosied** [comment]
Tests failed because of Java 21 used and it forced Java 8 to be the source .

`[3/7/24, 0:19:02:643 PST] 0000002b com.ibm.ws.jsp                                               I Requested jdkSourceLevel=17, but forcing to 18 because it is the min supported by Java 20+` 

Tested were expecting 17 so that the compilation would fail.

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+
+ *
+ */
+// No need to run against cdi-2.0 since these tests don't use CDI at all.
+@SkipForRepeat("CDI-2.0")
```
Comment (volosied) [inline dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:46]: @MinimumJavaLevel(javaLevel = 17)

Java 17 supports 7 still. 
```        Provide source compatibility with the specified Java SE release. Supported releases: 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17```

Comment (volosied) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:46]: We can also add test for 13, but we'd need to use Java 8 as the min.

File: dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (volosied) [reply dev/com.ibm.ws.jsp/resources/OSGI-INF/l10n/metatype.properties]: I reverted this change.  If we encounter any issues later on, we can always modify the wording later on.

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat

File: dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java (hunk 5 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:46]: Are you saying we need to add a min java level annotation?

Comment (volosied) [reply dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPJava7Test.java:46]: Correct. However, I don't see who would be using Java 1.3 at this point. Additionally, the behavior is already verified via these tests, so I don't see any need for another.

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
Oh!  Java 21 has Java 7 as the minimum, so which explains why the java 7 source using Java 8 lambda JSP passes when I expect it to fail. 

```
  --source <release>, -source <release>
        Provide source compatibility with the specified Java SE release.
        Supported releases: 
            8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21
```

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.jsp.2.3_fat/test-applications/TestJSPWithJava17.war/resources/testJava21.jsp [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<!--
+    Copyright (c) 2023 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.jsp.2.3_fat/test-applications/TestJSPWithJava17.war/resources/testJava21.jsp]: Is 2023 correct?

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**